zk_prover/src/chips/less_than_vertical.rs

```diff
@@ -0,0 +1,531 @@
+        });
+
+        meta.annotate_lookup_any_column(u8, || "LOOKUP_u8");
+
+        meta.lookup_any("range check for u8", |meta| {
+            let u8_cell = meta.query_advice(diff, Rotation::cur());
+            let u8_range = meta.query_fixed(u8, Rotation::cur());
+            let lookup_enable = meta.query_selector(lookup_enable);
+            vec![(lookup_enable * u8_cell, u8_range)]
+        });
+
+        LtVerticalConfig {
+            lt,
+            diff,
+            range,
+            u8,
+            lookup_enable,
+        }
+    }
+
+    /// Constructs a Lt chip given a config.
+    pub fn construct(config: LtVerticalConfig<N_BYTES>) -> LtVerticalChip<N_BYTES> {
+        LtVerticalChip { config }
+    }
+}
+
+impl<const N_BYTES: usize> LtVerticalInstruction for LtVerticalChip<N_BYTES> {
+    /// From lhs and rhs values, assigns `lt` and `diff_bytes` to the region.
+    fn assign(
+        &self,
+        region: &mut Region<'_, Fp>,
+        offset: usize,
+        lhs: Value<Fp>,
+        rhs: Value<Fp>,
+    ) -> Result<(), Error> {
+        let config = self.config();
+
+        let lt = lhs.zip(rhs).map(|(lhs, rhs)| lhs < rhs);
+
+        region.assign_advice(
+            || "lt chip: lt",
+            config.lt,
+            offset,
+            || lt.map(|lt| Fp::from(lt as u64)),
+        )?;
+
+        let diff_bytes = lhs.zip(rhs).map(|(lhs, rhs)| {
+            let mut diff = lhs - rhs;
+            let lt = lhs < rhs;
+            if lt {
+                diff += config.range;
+            } else {
+                diff += Fp::zero();
+            }
+            diff.to_repr()
+        });
+
+        for idx in 0..N_BYTES {
+            config.lookup_enable.enable(region, idx)?;
+
+            region.assign_advice(
+                || format!("lt chip: diff byte {}", idx),
+                config.diff,
+                offset + idx,
+                || diff_bytes.as_ref().map(|bytes| Fp::from(bytes[idx] as u64)),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    /// Loads the lookup table for `u8` range check.
+    fn load(&self, layouter: &mut impl Layouter<Fp>) -> Result<(), Error> {
+        const RANGE: usize = 256;
+
+        layouter.assign_region(
+            || "load u8 range check table",
+            |mut region| {
+                for i in 0..RANGE {
+                    region.assign_fixed(
+                        || "assign cell in fixed column",
+                        self.config.u8,
+                        i,
+                        || Value::known(Fp::from(i as u64)),
+                    )?;
+                }
+                Ok(())
+            },
+        )
+    }
+}
+
+impl<const N_BYTES: usize> Chip<Fp> for LtVerticalChip<N_BYTES> {
+    type Config = LtVerticalConfig<N_BYTES>;
+    type Loaded = ();
+
+    fn config(&self) -> &Self::Config {
+        &self.config
+    }
+
+    fn loaded(&self) -> &Self::Loaded {
+        &()
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
Let's have the tests in a separate file, as we do with other code in the project.

Reply:
Good point! I'll take it